### ISSUE TITLE
Add GraphQL endpoint to Sync

### ISSUE THREAD

**Issue by tcr** [tcr/edit-text#54]
Right now the Sync server (`mercutio-server/src/sync.rs`) and the file server (`mercutio-server/src/bin/mercutio-server.rs`) both share access to the database, which can result in spurious "database is locked" errors from SQLite. One solution to this problem is to make Sync have a GraphQL endpoint, and have the file server make modifications through this API.

As a result, we can also support a number of other features that fit an out-of-band request/response model:

* Downloading the raw Markdown or raw HTML for a file
* Overwriting a page with new Markdown contents
* Downloading an archive of all data
* Searching all documents for a string (full text search)
* Github or Dropbox login/logout

I played around with [Juniper](https://github.com/graphql-rust/juniper) and think it should be a great fit here.

**Comment by tcr** [comment]
A proof of concept would be:

1. Adding a function `sync_graphql_server()` to `mercutio-server/src/sync.rs` that suggests what port to expose itself
1. Having `sync_graphql_server` launch a basic Juniper graphql endpoint (some basic example like "hello world")
1. Adding a `sync_server()` function that calls `sync_socket_server()` and `sync_graphql_server()` in parallel, and changing `mercutio-server/src/bin/mercutio-server.rs` to call this parent method instead
1. [Configuring nginx](https://github.com/tcr/edit-text/blob/master/dist/deploy/nginx.conf) to reverse proxy this port on the `/$/graphql` endpoint
1. Adding a graphqlUrl() method to `mercutio-frontend/src/route.ts` to switch between local testing and nginx inside docker
1. Adding some example JavaScript into `mercutio-frontend/src/app.tsx` that calls that endpoint and returns the hello world string (but otherwise doesn't change the app)

The next step would be adding a proper GraphQL endpoint that can query the Markdown contents of an arbitrary page: `query { page(id: "home") { markdown } }`

**Comment by tcr** [comment]
The first implementation is now on master. 52a23ccfc1f0e8f7b8ca8607371bfac5d539d20a

**Comment by tcr** [comment]
A GraphQL server has been merged into `edit-server`, but no client-side code yet in `edit-frontend`.
